src/get_positional_features.py

```diff
@@ -0,0 +1,499 @@
+                'relative_page_number' : [],
+                'even_page' : [],
+                'year' : [],
+                'second_chamber' : [],
+                'unicameral' : [], 
+                'width' : [],
+                'height' : []}
+    
+    archive = pydl.LazyArchive()
+    protocols = sorted(list(protocol_iterators(args.records_folder, start=args.start, end=args.end)))
+    
+    curr_year = protocols[0].split('\\')[-2]
+    for protocol in progressbar.progressbar(protocols):
+        next_year = protocol.split('\\')[-2]
+        if curr_year != next_year:
+            output_df = pd.DataFrame.from_dict(feature_dict)
+            save_file = curr_year + '_position_features.csv'
+            output_df.to_csv(save_file, index = False)
+
+            curr_year = next_year
+            
+            feature_dict = {'id': [],
+                'record': [],
+                'page_number': [],
+                'page_offset' : [],
+                'posLeft' : [],
+                'posUpper' : [],
+                'posRight' : [],
+                'posLower' : [],
+                'relative_page_number' : [],
+                'even_page' : [],
+                'year' : [],
+                'second_chamber' : [],
+                'unicameral' : [], 
+                'width' : [],
+                'height' : []}
+            
+        # get xml position features
+        protocol_feature_dict = get_positional_features(protocol)
+        
+        # add coordinate data
+        protocol_feature_dict = add_coord_to_dict(protocol, protocol_feature_dict, archive)
+
+        # add to feature dict
+        record_file = protocol.split('\\')[-1]
+        feature_dict['id'].extend(protocol_feature_dict['id'])
+        feature_dict['record'].extend([record_file for x in range(len(protocol_feature_dict['id']))])
+        feature_dict['page_number'].extend(protocol_feature_dict['page_number'])
+        feature_dict['posLeft'].extend(protocol_feature_dict['posLeft'])
+        feature_dict['posUpper'].extend(protocol_feature_dict['posUpper'])
+        feature_dict['posRight'].extend(protocol_feature_dict['posRight'])
+        feature_dict['posLower'].extend(protocol_feature_dict['posLower'])
+        feature_dict['relative_page_number'].extend(protocol_feature_dict['relative_page_number'])
+        feature_dict['even_page'].extend(protocol_feature_dict['even_page'])
+        feature_dict['year'].extend(protocol_feature_dict['year'])
+        feature_dict['second_chamber'].extend(protocol_feature_dict['second_chamber'])
+        feature_dict['unicameral'].extend(protocol_feature_dict['unicameral'])
+        feature_dict['width'].extend(protocol_feature_dict['width'])
+        feature_dict['height'].extend(protocol_feature_dict['height'])
+        feature_dict['page_offset'].extend(protocol_feature_dict['page_offset'])
+    
+    # store features in dataframe and save to disk
+    save_file = curr_year + '_position_features.csv'
```

Comment:
If we do want to save the thing as a CSV; you should definitely have it in the args. Otherwise we will never know where the file is.